### ISSUE TITLE
Float with "in" operator

### ISSUE THREAD

**Issue by fmarcia** [antonmedv/expr#420]
With:
```go
code := "metric in 1..3"
env := map[string]any{"metric": float64(1)}
```
`Compile` raises an error:
```go
_, err := expr.Compile(code, expr.Env(env))
fmt.Println("compile:", err)
```
```
compile: cannot use float64 as type int in array (1:8)
 | metric in 1..3
 | .......^
```
But `Eval` works fine:
```go
_, err := expr.Eval(code, env)
fmt.Println("eval:", err)
```
```
eval: <nil>
```

[The code is here](https://goplay.tools/snippet/WGxIqtFwW30).

**Comment by antonmedv** [comment]
This is the purpose of compile.

In compile expr knows what float64 can't be in 1..3. But in Eval mode, metric has `any` type, so it is successfully executed.

**Comment by antonmedv** [comment]
Actually, this is strange behavior in Eval. Why nil?

**Comment by fmarcia** [comment]
> Actually, this is strange behavior in Eval. Why nil?

nil is the value of err

**Comment by antonmedv** [comment]
I think I got it, This is because of an optimization:

`metric in 1..3` changed to `1 <= metric && metric  <= 3`

**Comment by antonmedv** [comment]
```
package main

import (
	"fmt"

	"github.com/antonmedv/expr"
)

func main() {

	code := "metric in 1..3"
	env := map[string]any{"metric": float64(1.2)}

	program, err := expr.Compile(code, expr.Optimize(true))
	fmt.Println("compile:", err)

	out, err := expr.Run(program, env)
	fmt.Println("out:", out)
	fmt.Println("err:", err)
}

```

**Comment by antonmedv** [comment]
So this optimization should only be applied if `metric` has `int` type. Will add this check.

**Comment by antonmedv** [comment]
Fixed in e81f517f81a6fc91fc01134b7f76f731f1f3eb52
